### ISSUE TITLE
ScopedStorage in Api 30

### ISSUE THREAD

**Issue by michaelsam94** [esafirm/android-image-picker#361]
## Expected Behavior
In android API 30 write to external storage permission is ignored and 

## Actual Behavior
the write and read to external storage should be granted 

## Steps to Reproduce the Problem

  1.install app on API version 30 
  2.choose any picture 
  3.the permission to access file is denied 

## Specifications

  - Image Picker Version: 2.4.1
  - Android OS: 30
  - Phone: pixel 2

**Comment by michaelsam94** [comment]
I used both read and write to external storage but the recommended way to get image from the gallery is using intent you can specify the type of file you need like png or jpeg and launch intent to and put mime type you need and this intent will fire and let you select the document and the returned Uri is content Uri you need to convert it to the real path of the file

**Comment by esafirm** [comment]
@michaelsam94 honestly I'm kinda losing track on this thing. If we use ContentResolver to load the files, do we still need the permissions?

**Comment by xpdlf1004** [comment]
@esafirm Read permission need to access to the entire external storage. 
How about picking this commit to 3.x.
https://github.com/esafirm/android-image-picker/pull/362

**Comment by esafirm** [comment]
@xpdlf1004 I can't reproduce the issue in the emulator on 3.x
Can you please provide the sample of it? I don't have any device with OS 11 installed at the time.

**Comment by AnandRH** [comment]
Just use emulator with Android 11 and targetSDK to 30.
Storage premission will be no more shown.

**Comment by xpdlf1004** [comment]
@esafirm Oh, I found that situation! 

In a document for Android 11, It guides to only use "READ_EXTERNAL_STORAGE" for accessing the photo library without "WRITE_EXTERNAL_STORAGE" permission which is deprecated on Android 11. 

When you write the manifest like
```
<uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
<uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
```
Requesting "READ_EXTERNAL_STORAGE" give the "WRITE_EXTERNAL_STORAGE" too.  

But, If you write the manifest like the documents suggesting on Android 11
```
<uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
<uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" minSDK=28 />

or

<uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
```
It will give you only READ_EXTERNAL_STORAGE permission only without "WRITE" permission.

Because our library is checking the permission via "WRITE_EXTERNAL_STORAGE" on the loadDataWithPermission function, Someone using Android 11 and having a "READ" permission only(because of the above situations) can't use this library.

So I changed the "WRITE_EXTERNAL_STORAGE" permission to "READ" on "https://github.com/esafirm/android-image-picker/pull/362". 

If you want to reproduce this issue, write "READ_EXTERNAL_STORAGE" without "WRITE" on the manifest file. And use this library.
